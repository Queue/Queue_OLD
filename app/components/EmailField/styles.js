//
// EmailField styles.js

import { StyleSheet } from 'react-native'
import Colors from '../../lib/colors';
import Fonts from '../../lib/fonts';

export default styles = StyleSheet.create({
  emailField: {
    fontFamily: Fonts.content,
    margin: 10,
    height: 40,
    paddingLeft: 15,
    paddingRight: 15,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 2,
    backgroundColor: 'white'
  }
});
