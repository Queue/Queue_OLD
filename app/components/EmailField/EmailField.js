//
// Email Field Component

import React from 'react'
import { View, TextInput } from 'react-native';
import styles from './styles';
import Colors from '../../lib/colors';

export default EmailField = ({placeholder, change, val, focus, textColor}) => {

  // use the same styles object to add prop
  // changes so it can be controlled by state
  // in other components
  styles.props = {
    color: textColor
  };

  return (
    <TextInput
      style={ [styles.emailField,  styles.props] }
      autoFocus={focus}
      autoCapitalize={'none'}
      onChangeText={change}
      keyboardType={'email-address'}
      placeholder={placeholder}
      value={val}
    />
  );
}

EmailField.propTypes = {
  placeholder: React.PropTypes.string,
  val: React.PropTypes.string,
  change: React.PropTypes.func,
  textColor: React.PropTypes.string
};

EmailField.defaultProps = {
  placeholder: 'placeholder',
  textColor: Colors.primaryForeground
};
