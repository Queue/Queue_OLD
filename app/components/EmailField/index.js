//
// Email index.js

import EmailField from './EmailField';
export default EmailField;
