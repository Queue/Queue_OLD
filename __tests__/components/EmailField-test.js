//
// EmailField component test

import React from 'react';
import EmailField from '../../app/components/EmailField';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <EmailField
      placeholder='test'
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
