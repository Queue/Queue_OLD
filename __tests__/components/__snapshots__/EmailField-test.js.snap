exports[`test renders correctly 1`] = `
<TextInput
  autoCapitalize="none"
  autoFocus={undefined}
  keyboardType="email-address"
  onChangeText={undefined}
  placeholder="test"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 2,
        "fontFamily": "Helvetica Neue",
        "height": 40,
        "justifyContent": "center",
        "margin": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
      },
      Object {
        "color": "#48C948",
      },
    ]
  }
  value={undefined} />
`;
